Add a description of Step 2 (#793)
risc0/risc0: examples/factors/README.md

## Step 2: Give the methods package a name

The methods package contains the program that executes on the guest zkVM as well as a few supporting libraries. Before we proceed, let's change its name in `methods/Cargo.toml` to match the name of our project, renaming `name = methods` to `name = factors-methods`.

> Also `host/Cargo.toml` should be renamed `methods = { path = "../methods" }` to `factors-methods = { path = "../methods" }` in order to locate the renamed guest code.

Reviewer: One more wording revision.

```suggestion
Also in `host/Cargo.toml`, rename `methods = { path = "../methods" }` to `factors-methods = { path = "../methods" }` in order to locate the renamed guest code.
```
